qemu/tests/cdrom.py

```diff
@@ -359,21 +363,26 @@ def get_match_cdrom(session, serial_num):
         :return match_cdrom: the cdrom in guest which is corresponding
                              with the CML according to the serial number.
         """
+        def get_cdrom_by_serial(session, serial_num):
+            """
+            Get match cdrom from disk list, if not found return None.
+
+            :param session: VM session
+            :param serial_num: cdrom serial number
+            """
+            func = utils_misc.get_linux_drive_path
+            cdrom = utils_misc.wait_for(lambda: func(session, serial_num),
+                                        first=1, timeout=90)
+            return cdrom
+
         error.context("Get matching cdrom in guest", logging.info)
-        show_serial_num = "ls -l /dev/disk/by-id"
-        serial_num_output = session.cmd_output(show_serial_num)
-        if serial_num_output:
-            serial_cdrom = ""
-            for line in serial_num_output.splitlines():
-                if utils_misc.find_substring(str(line), str(serial_num)):
-                    serial_cdrom = line.split(" ")[-1].split("/")[-1]
-                    break
-            if not serial_cdrom:
-                qtree_info = vm.monitor.info("qtree")
-                raise error.TestFail("Could not find the device whose "
-                                     "serial number %s is same in Qemu"
-                                     " CML.\n Qtree info: %s" %
-                                     (qtree_info, serial_num))
+        serial_cdrom = get_cdrom_by_serial(session, serial_num)
+        if not serial_cdrom:
+            qtree_info = vm.monitor.info("qtree")
+            raise error.TestFail("Could not find the device whose "
+                                 "serial number %s is same in Qemu"
+                                 " CML.\n Qtree info: %s" %
+                                 (qtree_info, serial_num))
```

Comment:
And could you please add some debug logs at error ?
            raise error.TestFail("Could not find the corresponding cdrom"
                                 "in guest which is same in Qemu CML.")